table/tables/partition.go

```diff
@@ -233,14 +286,7 @@ func generateHashPartitionExpr(ctx sessionctx.Context, pi *model.PartitionInfo,
 
 // PartitionExpr returns the partition expression.
 func (t *partitionedTable) PartitionExpr() (*PartitionExpr, error) {
-	pi := t.meta.GetPartitionInfo()
-	switch pi.Type {
-	case model.PartitionTypeHash:
-		return t.partitionExpr, nil
-	case model.PartitionTypeRange:
-		return t.partitionExpr, nil
-	}
-	panic("cannot reach here")
+	return t.partitionExpr, nil
```

Comment:
Why change this?

Reply:
Because the old code are redundancy.
The code now are actual works the same as before, but cleaner.